### PR TITLE
Adopt Burst for parameterized tests

### PR DESCRIPTION
**Pull request by swankjesse**
It supports Kotlin/Multiplatform.

https://github.com/cashapp/burst

### CODE REVIEW

**Comment by swankjesse** [comment]
Looks good.
Screenshot 2024-10-16 at 9 35 02 AM

File: okio/src/jvmTest/kotlin/okio/BufferedSinkTest.kt [hunk 1]
```diff
@@ -27,27 +28,23 @@ import okio.TestUtil.segmentSizes
...
+@Burst
 class BufferedSinkTest(
   factory: Factory,
 ) {
-  interface Factory {
-    fun create(data: Buffer): BufferedSink
+  enum class Factory {
+    BUFFER {
+      override fun create(data: Buffer) = data
+      override fun toString() = "Buffer"
```
Comment (JakeWharton) [inline okio/src/jvmTest/kotlin/okio/BufferedSinkTest.kt:39]: We can rename all these in a follow-up to have these nicer names like you did for the new one at the bottom. No need for screaming case anymore.

**Review by JakeWharton — APPROVED** [review]
Nice. Just need to target the annotation to Java 8 and should be good.

File: okio/src/jvmTest/kotlin/okio/BufferedSinkTest.kt (hunk 1 above)
Comment (swankjesse) [reply okio/src/jvmTest/kotlin/okio/BufferedSinkTest.kt:39]: On it!

**Comment by swankjesse** [comment]
Hmmm, unexpected build failure:

```
> Task :okio:compileCommonMainKotlinMetadata FAILED
e: file:///Volumes/Development/okio/okio/src/commonMain/kotlin/okio/Buffer.kt:31:8 Class 'Buffer' is not abstract and does not implement abstract member 'timeout'.
e: file:///Volumes/Development/okio/okio/src/commonMain/kotlin/okio/HashingSink.kt:35:8 Class 'HashingSink' is not abstract and does not implement abstract member 'write'.
e: file:///Volumes/Development/okio/okio/src/commonMain/kotlin/okio/HashingSource.kt:36:8 Class 'HashingSource' is not abstract and does not implement abstract member 'read'.
e: file:///Volumes/Development/okio/okio/src/commonMain/kotlin/okio/RealBufferedSink.kt:19:17 Class 'RealBufferedSink' is not abstract and does not implement abstract member 'buffer'.
e: file:///Volumes/Development/okio/okio/src/commonMain/kotlin/okio/RealBufferedSource.kt:19:17 Class 'RealBufferedSource' is not abstract and does not implement abstract member 'buffer'.
```

**Comment by swankjesse** [comment]
Ooooh Kotlin 2. Checks out.
https://github.com/square/okio/pull/1535

**Pull request merged**